### PR TITLE
Replace FOS_FORCEFILESYSTEM by FOS_SUPPORTSTREAMABLEITEMS  in Windows file picker.

### PR DESCRIPTION
**Pull request by riastradh-brave**
fix brave/brave-browser#3638

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [ ] macOS
  - [x] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [x] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
https://hackerone.com/reports/506906

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by riastradh-brave** [comment]
Note: This change _unconditionally_ disables a feature in the file picker where you can paste a URL into it.  If we want to make this feature available _outside_ windows with Tor, then it will take a little more work.

File: chromium_src/ui/shell_dialogs/execute_select_file_win.cc [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+#include "ui/shell_dialogs/execute_select_file_win.h"
+
+#include <shlobjidl.h>
+
+// Disable FOS_FORCEFILESYSTEM.  If we set FOS_FORCEFILESYSTEM, and
+// you type a URL into the file picker, Windows will download that URL
+// and return a pathname to the downloaded file, with network traffic
+// beyond our control.
+#undef FOS_FORCEFILESYSTEM
+#define FOS_FORCEFILESYSTEM 0
```
Comment (darkdh) [inline chromium_src/ui/shell_dialogs/execute_select_file_win.cc]: in the original report, reporter mentioned unsetting `FOS_FORCEFILESYSTEM` is insufficient, we also need to set `FOS_SUPPORTSTREAMABLEITEMS`. How about we define `FOS_FORCEFILESYSTEM` as `FOS_SUPPORTSTREAMABLEITEMS` for this file?

Comment (riastradh-brave) [reply chromium_src/ui/shell_dialogs/execute_select_file_win.cc]: Oops, so there is, I missed that.  Sounds reasonable.  We may also have to worry about https://bugs.chromium.org/p/chromium/issues/detail?id=939985.

**Comment by riastradh-brave** [comment]
We discussed this at the security confab and decided that since there's a Chromium bug that causes the entire tab to crash if you paste in a URL that gives a 404, this is probably not a high-value feature we need to be afraid of disabling.

**Comment by riastradh-brave** [comment]
This doesn't seem to fix the problem.  Ugh!

**Comment by bbondy** [comment]
Is this still valid? No updates for just under a month. Thanks.

**Comment by riastradh-brave** [comment]
This turns out not to fix the problem, so I'm going to close this until we come up with more brilliant ideas.

**Pull request closed without merging**